src/polyfills/submit-event.ts

```diff
@@ -1,44 +0,0 @@
-type FormSubmitter = HTMLElement & { form?: HTMLFormElement; type?: string }
-
-const submittersByForm: WeakMap<HTMLFormElement, HTMLElement> = new WeakMap()
-
-function findSubmitterFromClickTarget(target: EventTarget | null): FormSubmitter | null {
-  const element = target instanceof Element ? target : target instanceof Node ? target.parentElement : null
-  const candidate = element ? (element.closest("input, button") as FormSubmitter | null) : null
-  return candidate?.type == "submit" ? candidate : null
-}
-
-function clickCaptured(event: Event) {
-  const submitter = findSubmitterFromClickTarget(event.target)
-
-  if (submitter && submitter.form) {
-    submittersByForm.set(submitter.form, submitter)
-  }
-}
-
-;(function () {
-  if ("submitter" in Event.prototype) return
-
-  let prototype = window.Event.prototype
-  // Certain versions of Safari 15 have a bug where they won't
-  // populate the submitter. This hurts TurboDrive's enable/disable detection.
-  // See https://bugs.webkit.org/show_bug.cgi?id=229660
-  if ("SubmitEvent" in window && /Apple Computer/.test(navigator.vendor)) {
-    prototype = window.SubmitEvent.prototype
-  } else if ("SubmitEvent" in window) {
-    return // polyfill not needed
-  }
-
-  addEventListener("click", clickCaptured, true)
```

Comment:
On a related note, capturing a `click` event for a submission **does not** polyfill the built-in behavior for calls to `HTMLFormElement.requestSubmit()` with a `submitter` argument. For example:

```html
<form>
  <button id="the-submitter">Submit</button>
  <button type="reset" id="not-the-submitter">Reset</button>
</form>

<script>
  const form = document.querySelector("form")
  const submitter = document.getElementById("the-submitter")

  form.requestSubmit(submitter)
</script>
```

In this case, the `submitter` argument is ignored, since it's never clicked and will not dispatch a `click` event.